examples/embeddable_examples/public/react_embeddables/search/search_react_embeddable.tsx

```diff
@@ -38,67 +37,82 @@ export const getSearchEmbeddableFactory = (services: Services) => {
 
       const api = buildApi(
         {
-          ...timeRangeApi,
+          ...timeRange.api,
           dataViews: dataViews$,
           dataLoading: dataLoading$,
           serializeState: () => {
             return {
               rawState: {
-                ...serializeTimeRange(),
+                ...timeRange.serialize(),
               },
               references: [],
             };
           },
         },
         {
-          ...timeRangeComparators,
+          ...timeRange.comparators,
         }
       );
 
+      let isUnmounted = false;
```

Comment:
nit: If you're only using this `isUnmounted` check to determine whether or not you should update your subjects, I'm not sure that's necessary.

I think the `useBatchedPublishingSubjects` and `useStateFromPublishingSubject` hooks should unsubscribe on unmount which prevents errors relating to setting state on unmounted components.

If you remove this, are there any warnings?